bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/ContentVariableTransformer.java

```diff
@@ -0,0 +1,145 @@
+    private PropertyAggregatorService aggregatorService;
+    private PropertyConfigService propertyConfigService;
+
+    public ContentVariableTransformer() {
+    }
+
+    public ContentVariableTransformer(PropertyAggregatorService propertyAggregatorService,
+                                      PropertyConfigService propertyConfigService) {
+        this.aggregatorService = propertyAggregatorService;
+        this.propertyConfigService = propertyConfigService;
+    }
+
+    @Override
+    public void init(ProcessingContext processingContext, ProcessingComponentConfiguration processingComponentConfiguration) throws IOException {
+        SlingHttpServletRequest request = processingContext.getRequest();
+
+        contentVariableReplacements = aggregatorService.getProperties(request);
+    }
+
+    public void startElement(String uri, String localName, String quaName, Attributes atts) throws SAXException {
+        if (shouldRun()) {
+            AttributesImpl newAttrs = new AttributesImpl(atts);
+            for (int i = 0; i < newAttrs.getLength(); i++) {
+                String currentAttribute = newAttrs.getValue(i);
+                final List<String> keys = ContentVariableReplacementUtil.getKeys(currentAttribute);
+                for (String key : keys) {
+                    // If the placeholder key is in the map then replace it
+                    if (ContentVariableReplacementUtil.hasKey(contentVariableReplacements, key)) {
+                        String replaceValue =
+                                (String) ContentVariableReplacementUtil.getValue(contentVariableReplacements, key);
+                        String newAttrValue = ContentVariableReplacementUtil.doReplacement(currentAttribute, key,
+                                baseEscaping(replaceValue), propertyConfigService.getAction(key));
+                        newAttrs.setValue(i, newAttrValue);
+                    }
+                }
+            }
+            getContentHandler().startElement(uri, localName, quaName, newAttrs);
+        } else {
+            getContentHandler().startElement(uri, localName, quaName, atts);
+        }
+    }
+
+    public void characters(char[] ch, int start, int length) throws SAXException {
+        String currentString = new String(ch, start, length);
+
+        if (shouldRun()) {
+
+            // Get the current placeholders in the string
+            final List<String> keys = ContentVariableReplacementUtil.getKeys(currentString);
+            for (String key : keys) {
+                // If the placeholder key is in the map then replace it
+                if (ContentVariableReplacementUtil.hasKey(contentVariableReplacements, key)) {
+                    final String placeholderReplacement =
+                            String.valueOf(ContentVariableReplacementUtil.getValue(contentVariableReplacements, key));
+                    currentString = ContentVariableReplacementUtil.doReplacement(currentString, key,
```

Comment:
Do we care that for URLs the ", ', <, and > characters will all be translated first to &quot; &apos; &lt; &gt; prior to then being encoded in the URL?  If those are actually ever in a valid value, we probably would want them to be URL encoded first right?

Reply:
I'm fine leaving it where it is once i make the change to enable disabling this escaping on a per-action basis

Reply:
ok, so you're thinking the UrlEncodeAction will be a type of action that disables escaping, b/c it's own escaping will be sufficient, correct?

Reply:
yeah that's my thought around it. this can stay here and it becomes a choice to do it or not in the Action (and yes I'll have to update UrlEncodeAction too)

Reply:
this should be addressed now - escaping happens on the output of the action unless explicitly skipped